src/Stratis.FederatedPeg.Features.FederationGateway/TemplateTransaction.cs

```diff
@@ -0,0 +1,21 @@
+﻿using NBitcoin;
+
```

Comment:
Not sure you need a backing property in there? Maybe I am wrong :/ is it something DBreeze specific ?

Reply:
Also wondering if we don't need some form of status stored in the database, like
pending-signature, in-mempool, on-chain, matured-on-chain, rejected, etc.
Or do you plan to put that in another object like "transfer" which would reference these templates ?